CanHazFunny/CanHazFunny/Jester.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace CanHazFunny;
+
+public class Jester
+{
+    private readonly IJokeService _jokeService;
+    private readonly IOutput _output;
+
+    public Jester(IOutput? output, IJokeService? jokeService)
+    {
+        _output = output ?? throw new ArgumentNullException(nameof(output));
+        _jokeService = jokeService ?? throw new ArgumentNullException(nameof(jokeService));
+        _jokeService = jokeService ?? throw new ArgumentNullException(nameof(jokeService));
+    }
+
+    public void TellJoke()
+    {
+        string joke;
+        while ((joke = _jokeService.GetJoke()).ToLower().Contains("chuck norris")) {}
```

Comment:
```suggestion
        string joke = _jokeService.GetJoke();
        while (jokeToLower().Contains("chuck norris")) {
          joke = _jokeService.GetJoke();
        }
```

While what you're doing works, I'd argue this is easier to read and understand.